src/main/java/sg/ncl/MainController.java

```diff
@@ -1319,45 +1326,81 @@ public String teams(Model model, HttpSession session) {
-                    image.setDescription(imageJsonObject.getString("description"));
-                    image.setTeamId(imageJsonObject.getString("teamId"));
-                    savedImageList.add(image);
-                }
-            }
+            imageMap.put(team2.getName(), invokeAndGetImageList(teamId));
         }
 
-        // get public teams
-//        HttpEntity<String> teamRequest = createHttpEntityHeaderOnly();
-//        ResponseEntity teamResponse = restTemplate.exchange(properties.getTeamsByVisibility(TeamVisibility.PUBLIC.toString()), HttpMethod.GET, teamRequest, String.class);
-//        String teamResponseBody = teamResponse.getBody().toString();
-//
-//        JSONArray teamPublicJsonArray = new JSONArray(teamResponseBody);
-//        for (int i = 0; i < teamPublicJsonArray.length(); i++) {
-//            JSONObject teamInfoObject = teamPublicJsonArray.getJSONObject(i);
-//            Team2 team2 = extractTeamInfo(teamInfoObject.toString());
-//            teamManager2.addTeamToPublicTeamMap(team2);
-//        }
+        // check if inner image map is empty, have to do it via this manner
+        // returns true if the team contains an image list
+        boolean isInnerImageMapPresent = imageMap.values().stream().filter(perTeamImageMap -> !perTeamImageMap.isEmpty()).findFirst().isPresent();
 
         model.addAttribute("userEmail", userEmail);
         model.addAttribute("teamMap2", teamManager2.getTeamMap());
-//        model.addAttribute("publicTeamMap2", teamManager2.getPublicTeamMap());
         model.addAttribute("userJoinRequestMap", teamManager2.getUserJoinRequestMap());
-        model.addAttribute("savedImageList", savedImageList);
+        model.addAttribute("isInnerImageMapPresent", isInnerImageMapPresent);
+        model.addAttribute("imageMap", imageMap);
         return "teams";
     }
 
+    /**
+     * Exectues the service-image and returns a Map containing the list of images in two partitions.
+     * One partition contains the list of already created images.
+     * The other partition contains the list of currently saving in progress images.
+     * @param teamId The ncl team id to retrieve the list of images from.
+     * @return Returns a Map containing the list of images in two partitions.
+     */
+    private Map<String, List<Image>> invokeAndGetImageList(String teamId) {
+        log.info("Getting list of saved images for team {}", teamId);
+
+        Map<String, List<Image>> resultMap = new HashMap<>();
+        List<Image> createdImageList = new ArrayList<>();
+        List<Image> inProgressImageList = new ArrayList<>();
+
+        HttpEntity<String> imageRequest = createHttpEntityHeaderOnly();
+        ResponseEntity imageResponse;
+        try {
+            imageResponse = restTemplate.exchange(properties.getTeamSavedImages(teamId), HttpMethod.GET, imageRequest, String.class);
+        } catch (ResourceAccessException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)